### PR TITLE
[RLlib] Cleanup `AlgorithmConfig.get_marl_module_spec()`.

### PR DESCRIPTION
**Pull request by sven1977**
The current implementation of `AlgorithmConfig.get_marl_module_spec()` is quite messy and needs cleanup and clarification via better comments.

Also, a simple test as follows would fail due to the logic in the method not pulling the default ModuleSpec where it would make sense:

```
config = (
    PPOConfig()
    .environment(
        observation_space=gym.spaces.Box(-1.0, 0.0, (64, 64, 3), np.float32),
        action_space=gym.spaces.Discrete(6),
    )
)
# Create our RLModule to compute actions with.
policy_dict, _ = config.get_multi_agent_setup()
module_spec = config.get_marl_module_spec(
    policy_dict=policy_dict,
    single_agent_rl_module_spec=None,  # <- None is the default here
)
module_spec.build()
```

Fails with:
```
File "/Users/sven/opt/anaconda3/envs/ray/lib/python3.8/site-packages/ray/rllib/algorithms/algorithm_config.py", line 3155, in get_marl_module_spec
    cur_marl_module_spec.module_specs, SingleAgentRLModuleSpec
AttributeError: 'NoneType' object has no attribute 'module_specs'
```

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm.py [hunk 1]
```diff
@@ -719,8 +719,6 @@ def setup(self, config: AlgorithmConfig) -> None:
             self.learner_group = learner_group_config.build()
 
             # check if there are modules to load from the module_spec
-            marl_module_ckpt_dir = None
```
Comment (sven1977) [inline rllib/algorithms/algorithm.py:722]: these lines are not needed, variables are re-defined directly hereafter.

File: rllib/algorithms/algorithm_config.py [hunk 2]
```diff
@@ -3105,7 +3106,7 @@ def get_marl_module_spec(
         self,
         *,
         policy_dict: Dict[str, PolicySpec],
-        module_spec: Optional[SingleAgentRLModuleSpec] = None,
+        single_agent_rl_module_spec: Optional[SingleAgentRLModuleSpec] = None,
```
Comment (sven1977) [inline rllib/algorithms/algorithm_config.py:3109]: Make this 100% explicit. There is some confusion when working with these specs as to which type is expected in which situation.

File: rllib/algorithms/algorithm_config.py [hunk 3]
```diff
@@ -3216,15 +3241,17 @@ def get_marl_module_spec(
 
         # Fill in the missing values from the specs that we already have. By combining
         # PolicySpecs and the default RLModuleSpec.
-        default_spec = self.get_default_rl_module_spec()
```
Comment (sven1977) [inline rllib/algorithms/algorithm_config.py:3219]: Already called above. No need to call this twice.

File: rllib/core/rl_module/marl_module.py [hunk 4]
```diff
@@ -84,8 +84,8 @@ def __check_module_configs(cls, module_configs: Dict[ModuleID, Any]):
                 )
 
     def keys(self) -> Iterator[ModuleID]:
-        """Returns an iteratable of module ids."""
-        return self._rl_modules.keys()
+        """Returns an iterable over the module IDs in this MultiAgentRLModule."""
+        return iter(self._rl_modules.keys())
```
Comment (sven1977) [inline rllib/core/rl_module/marl_module.py:88]: Small fix. keys() returns a generator, not an iterator :)

Comment (kouroshHakha) [reply rllib/core/rl_module/marl_module.py:88]: We probably have to fix the type. My intention was to return what ever the dict.keys() returns. So it's readable via pdb.

**Review by kouroshHakha — APPROVED** [review]
Looks good. Just one small change you made is actually not what the intention was. We need to fix it by changing the type hint. Approved conditioned on fixing that one.

File: rllib/core/rl_module/marl_module.py (hunk 4 above)
Comment (sven1977) [reply rllib/core/rl_module/marl_module.py:88]: Ah, got it. Thanks for the explanation @kouroshHakha ! Will fix the typehint to generator and revert the above change.

Comment (sven1977) [reply rllib/core/rl_module/marl_module.py]: done: Changed to KeysView, which is the type that a dict.keys() would return
¯\_(ツ)_/¯

**Pull request merged**